lightning/src/offers/flow.rs

```diff
@@ -1082,4 +1133,317 @@ where
 	) -> Vec<(DNSResolverMessage, MessageSendInstructions)> {
 		core::mem::take(&mut self.pending_dns_onion_messages.lock().unwrap())
 	}
+
+	/// Retrieve an [`Offer`] for receiving async payments as an often-offline recipient. Will only
+	/// return an offer if [`Self::set_paths_to_static_invoice_server`] was called and we succeeded in
+	/// interactively building a [`StaticInvoice`] with the static invoice server.
+	#[cfg(async_payments)]
+	pub(crate) fn get_async_receive_offer(&self) -> Result<(Offer, bool), ()> {
+		let mut cache = self.async_receive_offer_cache.lock().unwrap();
+		cache.get_async_receive_offer(self.duration_since_epoch())
+	}
+
+	/// Sends out [`OfferPathsRequest`] and [`ServeStaticInvoice`] onion messages if we are an
+	/// often-offline recipient and are configured to interactively build offers and static invoices
+	/// with a static invoice server.
+	///
+	/// # Usage
+	///
+	/// This method should be called on peer connection and once per minute or so, to keep the offers
+	/// cache updated. When calling this method once per minute, SHOULD set `timer_tick_occurred`.
+	///
+	/// Errors if we failed to create blinded reply paths when sending an [`OfferPathsRequest`] message.
+	#[cfg(async_payments)]
+	pub(crate) fn check_refresh_async_receive_offers<ES: Deref, R: Deref>(
```

Comment:
Does this name capture well enough that invoices are also refreshed if necessary?